sdk/servicebus/azure-servicebus/api_review.py

```diff
@@ -0,0 +1,588 @@
+        entity_name : str,
+        credential: TokenCredential,
+        subscription_name : str = None,
+        session : str = None,
+        mode : ReceiveSettleMode = ReceiveSettleMode.PeekLock
+        logging_enable: bool = False,
+        http_proxy: dict = None,
+        transport_type: TransportType = None,
+        retry_total : int = 3,
+    ) -> None:
+    @classmethod
+    def from_queue(
+        cls,
+        fully_qualified_namespace : str,
+        queue_name : str,
+        credential : TokenCredential,
+        session: str = None,
+        mode : ReceiveSettleMode = ReceiveSettleMode.PeekLock
+        **kwargs
+    ) -> ServiceBusReceiverClient:
+    @classmethod
+    def from_topic_subscription(
+        cls,
+        fully_qualified_namespace : str,
+        topic_name : str,
+        subscription_name : str,
+        credential: TokenCredential,
+        session: str = None,
+        mode : ReceiveSettleMode = ReceiveSettleMode.PeekLock
+        **kwargs
+    ) -> ServiceBusReceiverClient:
+    @classmethod
+    def from_connection_string(
+        cls,
+        conn_str : str,
+        entity_name : str = None,
+        subscription_name : str = None,
+        session: str = None,
+        mode : ReceiveSettleMode = ReceiveSettleMode.PeekLock
+        **kwargs
+    )-> ServiceBusReceiverClient:
+
+    def __enter__(self):
+    def __exit__(self):
+
+    def close(self) -> None:
+
+    def get_properties(self) -> Dict[str, any]:
+
+    def __iter__(self):
+    def __next__(self):
+    def next(self):
+
+    def peek(
+        self,
+        count : int = 1,
+        start_from : int = None
+    ) -> List[PeekMessage]:
+    def receive_deferred_messages(
+        self,
+        sequence_numbers : List[int],
+    ) -> List[DeferredMessage]:
+    def receive(
+        self,
+        max_batch_size : int = None,
+        timeout : float = None,
+    ) -> List[Message]:  # Pull mode receive
+    def settle_deferred_messages(
+        self,
+        settlement : str,  # TODO: In T1 settlement is just string like 'completed', 'suspended', 'abandoned', can we improve this parameter to be Enum or remove this method?
+        messages : List[DeferredMessage],
+        **kwargs
+    ) -> None:  # Batch settle deferred messages
+
+    def list_sessions(self) -> List[str]:
+    def get_session(self) -> Session:  # raise Error when called on non-session entity
+
+    # Rule APIs, raise Error when called on Queue
+    def add_rule(self, rule_name : str, filter : str) -> None: # TODO: figure out what filter is
+    def remove_rule(self, rule_name : str) -> None:
+    def get_rules(self) -> List[str] -> Dict[str, Any]:
+
+
+class Session:
+    def get_session_id(self) -> str:
+    def get_session_state(self) -> str:  # This can be made to property if property is more pythonic way
+    def set_session_state(self, state : str) -> None:  # This can be made to property if property is more pythonic way
```

Comment:
While get/set isn't a Pythonic naming pattern, in this case I would prefer not to use a property as it masks that fact that there's a service call involved, so I think we should keep the methods I think :)